### PR TITLE
DEPR: Deprecate pandas.np module

### LINKED ISSUE #30296: DEPR: Remove pandas.np
Not sure if it was added intentionally, but it's possible to call numpy with the `np` attribute of the pandas module:
```python
import pandas
x = pandas.np.array([1, 2, 3])
```
While this is not documented, I've seen couple of places suggesting this as a "trick" to avoid importing numpy directly.

I personally find this hacky, and I think should be removed.

### PR DESCRIPTION
**Pull request by lithomas1**
- [x] closes #30296 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/__init__.py [hunk 1]
```diff
@@ -211,6 +210,19 @@ class Panel:
                 pass
 
             return Panel
+
+        elif name == "np":
```
Comment (jbrockmendel) [inline pandas/__init__.py]: this only works for py37+, py36 needs to be handled separately

Comment (jreback) [reply pandas/__init__.py]: @jbrockmendel side issue - we should remove the Panel compat

**Comment by jorisvandenbossche** [comment]
> py36 needs to be handled separately

You can still do the `import numpy as np` in the `else` clause.

Can you also add tests for this? (that it still works but generates a warning for py37)

File: doc/source/whatsnew/v1.0.0.rst [hunk 2]
```diff
@@ -205,6 +205,7 @@ Other enhancements
   (:meth:`~DataFrame.to_parquet` / :func:`read_parquet`) using the `'pyarrow'` engine
   now preserve those data types with pyarrow >= 1.0.0 (:issue:`20612`).
 - The ``partition_cols`` argument in :meth:`DataFrame.to_parquet` now accepts a string (:issue:`27117`)
+- The pandas.np submodule is now deprecated. Import numpy directly instead(:issue:`30296`)
```
Comment (gfyoung) [inline doc/source/whatsnew/v1.0.0.rst]:
```suggestion
- The pandas.np submodule is now deprecated. Import numpy directly instead (:issue:`30296`)
```

**Comment by lithomas1** [comment]
@jbrockmendel and @jorisvandenbossche 
Can you take another look at the code? I added tests and made the deprecation warnings also show on Python 3.6

File: doc/source/whatsnew/v1.0.0.rst (hunk 2 above)
Comment (lithomas1) [reply doc/source/whatsnew/v1.0.0.rst]: fixed

File: pandas/__init__.py (hunk 1 above)
Comment (lithomas1) [reply pandas/__init__.py]: added py36 deprecation.

**Review by gfyoung — APPROVED** [review]

File: pandas/tests/api/test_api.py [hunk 3]
```diff
@@ -220,6 +221,22 @@ def test_api(self):
...
+    def test_depr(self):
+        deprecated = (
+            self.deprecated_modules
+            + self.deprecated_classes
+            + self.deprecated_classes_in_future
+            + self.deprecated_funcs
+            + self.deprecated_funcs_in_future
+        )
+        for depr in deprecated:
+            with tm.assert_produces_warning(FutureWarning, check_stacklevel=False):
```
Comment (jorisvandenbossche) [inline pandas/tests/api/test_api.py]: Is the `check_stacklevel=False` needed here?

Comment (jorisvandenbossche) [reply pandas/tests/api/test_api.py]: Also, I would maybe add a separate, specific test for the numpy deprecation as well, where eg you can call `pd.np.arange(10)` and check that it actually works.

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for the updates!

File: pandas/tests/api/test_api.py (hunk 3 above)
Comment (lithomas1) [reply pandas/tests/api/test_api.py]: fixed

**Review by datapythonista — APPROVED** [review]
lgtm

Cools stuff @lithomas1, thanks for working on this.

File: doc/source/whatsnew/v1.0.0.rst [hunk 4]
```diff
@@ -205,6 +205,7 @@ Other enhancements
   (:meth:`~DataFrame.to_parquet` / :func:`read_parquet`) using the `'pyarrow'` engine
   now preserve those data types with pyarrow >= 1.0.0 (:issue:`20612`).
 - The ``partition_cols`` argument in :meth:`DataFrame.to_parquet` now accepts a string (:issue:`27117`)
+- The pandas.np submodule is now deprecated. Import numpy directly instead (:issue:`30296`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.0.0.rst]: can you put ``pandas.np`` in double-backticks

**Review by jreback — APPROVED** [review]
@lithomas1 lgtm. very minor comment. pls rebase on master and ping on green.

**Comment by lithomas1** [comment]
@jreback 
CI has passed.

**Comment by jreback** [comment]
thanks @lithomas1 very nice!

File: pandas/__init__.py [hunk 5]
```diff
@@ -236,6 +248,28 @@ class SparseDataFrame:
...
+    class __numpy:
+        def __init__(self):
+            import numpy as np
+            import warnings
+
+            self.np = np
+            self.warnings = warnings
+
+        def __getattr__(self, item):
+            self.warnings.warn(
```
Comment (jbrockmendel) [inline pandas/__init__.py]: im getting a mypy complaint about this.  why use self.warnings rather than just warnings?

**Pull request merged**